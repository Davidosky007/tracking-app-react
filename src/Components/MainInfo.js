import PropTypes from 'prop-types';
import { icons, measurementUnits } from '../logic/info';

const MainInfo = (props) => {
  const {
    unit, measurements, getMeasurementsByDate, reduceMethod, selectedDate,
  } = props;
  const measurementsByDate = getMeasurementsByDate(measurements, selectedDate);
  const displayedValue = reduceMethod(measurementsByDate);

  return (
    <div className="main-info">
      <p className="big m-b-10 color-dark-gray">{displayedValue}</p>
      <i className={`${icons[unit]} x-small color-gray`} />
      <p className="small color-gray">{unit}</p>
      <p className="x-small color-gray">{measurementUnits[unit]}</p>
    </div>
  );
};

MainInfo.propTypes = {
  unit: PropTypes.string.isRequired,
  measurements: PropTypes.arrayOf(PropTypes.object).isRequired,
  getMeasurementsByDate: PropTypes.func.isRequired,
  reduceMethod: PropTypes.func.isRequired,
  selectedDate: PropTypes.shape({}).isRequired,
};

export default MainInfo;
