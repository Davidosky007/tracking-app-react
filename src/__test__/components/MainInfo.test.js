import renderer from 'react-test-renderer';
import MainInfo from '../../Components/MainInfo';

describe('MainInfo', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<MainInfo
      unit=""
      measurements={[{}]}
      getMeasurementsByDate={() => {}}
      reduceMethod={() => {}}
      selectedDate={{}}
    />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
