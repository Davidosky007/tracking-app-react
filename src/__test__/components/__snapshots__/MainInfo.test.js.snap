// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainInfo renders correctly 1`] = `
<div
  className="main-info"
>
  <p
    className="big m-b-10 color-dark-gray"
  />
  <i
    className="undefined x-small color-gray"
  />
  <p
    className="small color-gray"
  >
    
  </p>
  <p
    className="x-small color-gray"
  />
</div>
`;
